### PR TITLE
Support "select into outfile" clause

### PR DESCRIPTION
**Pull request by c1oudman**
For example:
```
SELECT * FROM customer limit 100
INTO OUTFILE "hdfs://127.0.0.1:9002/user/c1oudman/"
FORMAT AS CSV
PROPERTIES
(
    "broker.name" = "dorisdb",
    "broker.username" = "dorisdb",
    "column_separator" = ",",
    "line_delimiter" = "\n",
    "max_file_size" = "100MB"
);
```

### CODE REVIEW

File: be/src/exec/plain_text_builder.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    WritableFile* _writable_file;
+
+    // Used to buffer the export data of plain text
+    // TODO(cmy): I simply use a stringstrteam to buffer the data, to avoid calling
+    // file writer's write() for every single row.
+    // But this cannot solve the problem of a row of data that is too large.
+    // For exampel: bitmap_to_string() may return large volumn of data.
+    // And the speed is relative low, in my test, is about 6.5MB/s.
+    std::stringstream _plain_text_outstream;
```
Comment (kangkaisen) [inline be/src/exec/plain_text_builder.h:51]: Could we use `fmt`?

File: be/src/column/column.h [hunk 2]
```diff
@@ -300,6 +300,8 @@ class Column {
     // Push one row to MysqlRowBuffer
     virtual void put_mysql_row_buffer(MysqlRowBuffer* buf, size_t idx) const = 0;
 
+    virtual std::string to_string(size_t idx, const std::string& null_value) const { return null_value; }
```
Comment (kangkaisen) [inline be/src/column/column.h:303]: I think we need a more concrete name.

**Comment by imay** [comment]
@c1oudman 
We have supported csv decode in be/formats/csv, I think it's better to implement csv encode in it too.

File: be/src/column/binary_column.h [hunk 3]
```diff
@@ -206,6 +206,10 @@ class BinaryColumn final : public ColumnFactory<Column, BinaryColumn> {
 
     void put_mysql_row_buffer(MysqlRowBuffer* buf, size_t idx) const override;
 
+    std::string to_string(size_t idx, const std::string& null_value) const override {
+        return get_slice(idx).to_string();
```
Comment (kangkaisen) [inline be/src/column/binary_column.h:210]: Could directly  return new std::string, needn't get_slice.

**Comment by c1oudman** [comment]
@imay @kangkaisen PTAL

File: be/src/exec/plain_text_builder.cpp [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            auto col_ptr = columns_raw_ptr[col];
+            RETURN_IF_ERROR(_converters[col]->write_string(os, *col_ptr, row, opts));
+            RETURN_IF_ERROR(os->write((col == num_cols - 1) ? row_delimiter : column_delimiter));
+        }
+    }
+    return Status::OK();
+}
+
+std::size_t PlainTextBuilder::file_size() {
+    assert(_output_stream != nullptr);
```
Comment (stdpain) [inline be/src/exec/plain_text_builder.cpp:83]: prefer DCHECK instead of assert

File: be/src/exec/plain_text_builder.h [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+class ExprContext;
+class FileWriter;
+class RowBatch;
+
+struct PlainTextBuilderOptions {
+    std::string column_terminated_by;
+    std::string line_terminated_by;
+};
+
+class PlainTextBuilder : public FileBuilder {
```
Comment (kangkaisen) [inline be/src/exec/plain_text_builder.h:28]: Could mark `PlainTextBuilder` final.

File: be/src/exec/file_builder.h [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+#pragma once
+
+#include "env/env.h"
+#include "gen_cpp/DataSinks_types.h"
+#include "runtime/result_writer.h"
+#include "runtime/runtime_state.h"
+
+namespace starrocks {
+
+class FileBuilder {
```
Comment (kangkaisen) [inline be/src/exec/file_builder.h:12]: Would better add some comments for this class.

**Review by kangkaisen — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Pull request merged**